### PR TITLE
feat: update Referrers API for distribution-spec v1.1.0-rc3

### PR DESCRIPTION
**Pull request by Wwwsylvia**
Resolves: #443
Resolves: #533

### CODE REVIEW

File: registry/remote/repository.go [hunk 1]
```diff
@@ -47,11 +47,24 @@ import (
...
-// See https://github.com/opencontainers/distribution-spec/blob/v1.1.0-rc1/spec.md#pull
-const dockerContentDigestHeader = "Docker-Content-Digest"
+const (
+	// headerDockerContentDigest is the "Docker-Content-Digest" header.
+	// If present on the response, it contains the canonical digest of the
+	// uploaded blob.
+	//
+	// References:
+	//   - https://docs.docker.com/registry/spec/api/#digest-header
+	//   - https://github.com/opencontainers/distribution-spec/blob/v1.1.0-rc1/spec.md#pull
```
Comment (shizhMSFT) [inline registry/remote/repository.go:57]: We should create an issue to update all reference to `rc1` to `rc3` to check if it is update to date.

File: registry/remote/repository.go [hunk 2]
```diff
@@ -497,10 +510,19 @@ func (r *Repository) referrersPageByAPI(ctx context.Context, artifactType string
 	if err := json.NewDecoder(lr).Decode(&index); err != nil {
 		return "", fmt.Errorf("%s %q: failed to decode response: %w", resp.Request.Method, resp.Request.URL, err)
 	}
+
 	referrers := index.Manifests
-	if artifactType != "" && !isReferrersFilterApplied(index.Annotations, "artifactType") {
-		// perform client side filtering if the filter is not applied on the server side
-		referrers = filterReferrers(referrers, artifactType)
+	if artifactType != "" {
+		requested := "artifactType"
```
Comment (shizhMSFT) [inline registry/remote/repository.go:516]: Should we make it a `const`?

File: registry/remote/repository.go (hunk 1 above)
Comment (Wwwsylvia) [reply registry/remote/repository.go:57]: Issue created: #554

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**